test/e2e/kustomize_provisioner_test.go

```diff
@@ -0,0 +1,315 @@
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			err := c.Create(ctx, bd)
+			Expect(err).To(BeNil())
+		})
+		AfterEach(func() {
+			By("deleting the testing resources")
+			Expect(c.Delete(ctx, bd)).To(BeNil())
+		})
+
+		It("should rollout the bundle contents successfully", func() {
+			By("eventually writing a successful installation state back to the bundledeployment status")
+			Eventually(func() (*metav1.Condition, error) {
+				if err := c.Get(ctx, client.ObjectKeyFromObject(bd), bd); err != nil {
+					return nil, err
+				}
+				if bd.Status.ActiveBundle == "" {
+					return nil, fmt.Errorf("waiting for bundle name to be populated")
+				}
+				return meta.FindStatusCondition(bd.Status.Conditions, rukpakv1alpha1.TypeInstalled), nil
+			}).Should(And(
+				Not(BeNil()),
+				WithTransform(func(c *metav1.Condition) string { return c.Type }, Equal(rukpakv1alpha1.TypeInstalled)),
+				WithTransform(func(c *metav1.Condition) metav1.ConditionStatus { return c.Status }, Equal(metav1.ConditionTrue)),
+				WithTransform(func(c *metav1.Condition) string { return c.Reason }, Equal(rukpakv1alpha1.ReasonInstallationSucceeded)),
+				WithTransform(func(c *metav1.Condition) string { return c.Message }, ContainSubstring("Instantiated bundle")),
+			))
+		})
+
+		When("the generated resource contains a pod manifest", func() {
+			It("should eventually result in an available pod resource", func() {
+				By("eventually install generated resource successfully")
+				pod := &corev1.Pod{}
+
+				Eventually(func() (*corev1.PodCondition, error) {
+					if err := c.Get(ctx, types.NamespacedName{Name: "dev-myapp-pod", Namespace: defaultSystemNamespace}, pod); err != nil {
+						return nil, err
+					}
+					for _, c := range pod.Status.Conditions {
+						if c.Type == corev1.PodReady {
+							return &c, nil
+						}
+					}
+					return nil, nil
+				}).Should(And(
+					Not(BeNil()),
+					WithTransform(func(c *corev1.PodCondition) corev1.PodConditionType { return c.Type }, Equal(corev1.PodReady)),
+					WithTransform(func(c *corev1.PodCondition) corev1.ConditionStatus { return c.Status }, Equal(corev1.ConditionTrue)),
+				))
+			})
+
+			It("should re-create a deployment resource when manually deleted", func() {
+				pod := &corev1.Pod{}
+
+				Eventually(func() error {
+					return c.Get(ctx, types.NamespacedName{Name: "dev-myapp-pod", Namespace: defaultSystemNamespace}, pod)
+				}).Should(Succeed())
+
+				By("storing the pod's original UID")
+				originalUID := pod.GetUID()
+
+				By("deleting the underlying pod and waiting for it to be re-created")
+				err := c.Delete(context.Background(), pod)
+				Expect(err).To(BeNil())
+
+				By("verifying the pod's UID has changed")
+				Eventually(func() (types.UID, error) {
+					err := c.Get(ctx, client.ObjectKeyFromObject(pod), pod)
+					return pod.GetUID(), err
+				}).ShouldNot(Equal(originalUID))
+			})
+		})
+	})
+	When("a BundleDeployment targets a valid Bundle", func() {
```

Comment:
nit: this test case name mirrors the same name as the above test. If I'm reading the differences correctly, it looks like this test is geared towards nested kustomize paths?

Reply:
This test pulls the whole repo contents as bundle and specify the path in it instead of pulling only the kustomize directory contents and specifying the path in it.  I'll update the test case name.  Thanks!